### ISSUE TITLE
Exporting all MBeans?

### ISSUE THREAD

**Issue by jcf** [onyx-platform/onyx-peer-http-query#4]
I'm looking into upgrading from Onyx 0.9 to 0.10, and in the process of looking into how we'll get metrics working noticed what looks like [exporting all MBeans](https://github.com/onyx-platform/onyx-peer-http-query/blob/0baf08a95a849cc247b2996042c6acb614ec84de/src/onyx/metrics_endpoint.clj#L66) via the `/metrics` endpoint.

``` clj
(defn metrics-endpoint [peer-config]
  (let [builder (java.lang.StringBuilder.)
        blacklists (:onyx.query.server/metrics-blacklist peer-config)]
    (doseq [mbean (jmx/mbean-names "*:*")] 
      (doseq [attribute (jmx/attribute-names mbean)]
        (try 
         (let [value (jmx/read mbean attribute)] 
           (cond (number? value) 
                 (let [metric (job-metric->metric-str (.getCanonicalKeyPropertyListString mbean) attribute)]
                   (when-not (blacklisted? blacklists metric)
                     (.append builder (format "%s %s" metric value))
                     (.append builder "\n")))
                 (map? value) 
                 (run! (fn [[k v]]
                         (when (number? v)
                           (let [metric (job-metric->metric-str (.getCanonicalKeyPropertyListString mbean) 
                                                                (str (name attribute) "_" (name k)))]
                             (when-not (blacklisted? blacklists metric)
                               (.append builder (format "%s %s" metric v))
                               (.append builder "\n")))))
                       value)))
         ;; Safe to swallow
         (catch javax.management.RuntimeMBeanException _))))
    (str builder)))
```

Perhaps there's some filtering in the blacklist that I've not seen yet, but I wonder why there isn't a JMX domain of say `org.onyxplatform` to isolate your MBeans from the user's. You could load just yours then with `(jmx/mbean-names "org.onyxplatform:*")`, and not have to worry about exposing anyone else's data by accident.

If I've missed some filtering/sanitisation I apologise. I'm only just starting to familiarise myself with this code.

**Comment by lbradstreet** [comment]
Thanks @jcf. We should do better here. I originally couldn't figure out how to report our metrics namespaced, however we really should do so. I've tracked down how to do so with the JmxReporter, so this should make filtering easier. I'll have the improved namespacing mechanism and improved filters included in our next release.

Cheers,

Lucas

**Comment by jcf** [comment]
Awesome! Thanks, @lbradstreet.

**Comment by jcf** [comment]
Sorry, reopened as I've noticed you assigned yourself. :smile:

**Comment by lbradstreet** [comment]
This is solved in 0.10.0.0-beta11. See section in the README regarding
```
 :onyx.query.server/metrics-selectors ["org.onyxplatform:*" "com.amazonaws.management:*"]
```
